### ISSUE TITLE
Enabling health check causes webhook permission issue with Kubernetes 1.23

### ISSUE THREAD

**Issue by KannanThiru** [kubeops/config-syncer#500]
### Enabling health check causes webhook permission issue with Kubernetes 1.23

**Kubed Version: V0.12.0
Kubernetes Version: V1.23.0**

**I am able to reproduce the same with Minikube 1.24.0 with K8s 1.23.0**

```
  healthcheck:
    # healthcheck configures the readiness and liveliness probes for the operator pod.
    enabled: true
```

```
$ kubectl logs kubed-547774cdf6-ggdl8 -f 
I1215 09:26:09.135697       1 log.go:181] FLAG: --alsologtostderr="true"
I1215 09:26:09.136389       1 log.go:181] FLAG: --audit-dynamic-configuration="false"
I1215 09:26:09.136405       1 log.go:181] FLAG: --audit-log-batch-buffer-size="10000"
I1215 09:26:09.136410       1 log.go:181] FLAG: --audit-log-batch-max-size="1"
I1215 09:26:09.136415       1 log.go:181] FLAG: --audit-log-batch-max-wait="0s"
I1215 09:26:09.136419       1 log.go:181] FLAG: --audit-log-batch-throttle-burst="0"
I1215 09:26:09.136424       1 log.go:181] FLAG: --audit-log-batch-throttle-enable="false"
I1215 09:26:09.136429       1 log.go:181] FLAG: --audit-log-batch-throttle-qps="0"
I1215 09:26:09.136434       1 log.go:181] FLAG: --audit-log-format="json"
I1215 09:26:09.136438       1 log.go:181] FLAG: --audit-log-maxage="0"
I1215 09:26:09.136442       1 log.go:181] FLAG: --audit-log-maxbackup="0"
I1215 09:26:09.136454       1 log.go:181] FLAG: --audit-log-maxsize="0"
I1215 09:26:09.136459       1 log.go:181] FLAG: --audit-log-mode="blocking"
I1215 09:26:09.136464       1 log.go:181] FLAG: --audit-log-path="-"
I1215 09:26:09.136468       1 log.go:181] FLAG: --audit-log-truncate-enabled="false"
I1215 09:26:09.136473       1 log.go:181] FLAG: --audit-log-truncate-max-batch-size="10485760"
I1215 09:26:09.136479       1 log.go:181] FLAG: --audit-log-truncate-max-event-size="102400"
I1215 09:26:09.136484       1 log.go:181] FLAG: --audit-log-version="audit.k8s.io/v1"
I1215 09:26:09.136489       1 log.go:181] FLAG: --audit-policy-file=""
I1215 09:26:09.136494       1 log.go:181] FLAG: --audit-webhook-batch-buffer-size="10000"
I1215 09:26:09.136499       1 log.go:181] FLAG: --audit-webhook-batch-initial-backoff="10s"
I1215 09:26:09.136504       1 log.go:181] FLAG: --audit-webhook-batch-max-size="400"
I1215 09:26:09.136508       1 log.go:181] FLAG: --audit-webhook-batch-max-wait="30s"
I1215 09:26:09.136517       1 log.go:181] FLAG: --audit-webhook-batch-throttle-burst="15"
I1215 09:26:09.136521       1 log.go:181] FLAG: --audit-webhook-batch-throttle-enable="true"
I1215 09:26:09.136527       1 log.go:181] FLAG: --audit-webhook-batch-throttle-qps="10"
I1215 09:26:09.136532       1 log.go:181] FLAG: --audit-webhook-config-file=""
I1215 09:26:09.136537       1 log.go:181] FLAG: --audit-webhook-initial-backoff="10s"
I1215 09:26:09.136541       1 log.go:181] FLAG: --audit-webhook-mode="batch"
I1215 09:26:09.136549       1 log.go:181] FLAG: --audit-webhook-truncate-enabled="false"
I1215 09:26:09.136553       1 log.go:181] FLAG: --audit-webhook-truncate-max-batch-size="10485760"
I1215 09:26:09.136558       1 log.go:181] FLAG: --audit-webhook-truncate-max-event-size="102400"
I1215 09:26:09.136563       1 log.go:181] FLAG: --audit-webhook-version="audit.k8s.io/v1"
I1215 09:26:09.136568       1 log.go:181] FLAG: --authentication-kubeconfig=""
I1215 09:26:09.136572       1 log.go:181] FLAG: --authentication-skip-lookup="true"
I1215 09:26:09.136577       1 log.go:181] FLAG: --authentication-token-webhook-cache-ttl="***REDACTED***"
I1215 09:26:09.136582       1 log.go:181] FLAG: --authentication-tolerate-lookup-failure="false"
I1215 09:26:09.136589       1 log.go:181] FLAG: --authorization-always-allow-paths="[]"
I1215 09:26:09.136594       1 log.go:181] FLAG: --authorization-kubeconfig=""
I1215 09:26:09.136598       1 log.go:181] FLAG: --authorization-webhook-cache-authorized-ttl="10s"
I1215 09:26:09.136603       1 log.go:181] FLAG: --authorization-webhook-cache-unauthorized-ttl="10s"
I1215 09:26:09.136608       1 log.go:181] FLAG: --bind-address="0.0.0.0"
I1215 09:26:09.136613       1 log.go:181] FLAG: --burst="1000000"
I1215 09:26:09.136619       1 log.go:181] FLAG: --cert-dir="apiserver.local.config/certificates"
I1215 09:26:09.136623       1 log.go:181] FLAG: --client-ca-file=""
I1215 09:26:09.136628       1 log.go:181] FLAG: --cluster-name="unicorn"
I1215 09:26:09.136632       1 log.go:181] FLAG: --config-source-namespace=""
I1215 09:26:09.136636       1 log.go:181] FLAG: --contention-profiling="false"
I1215 09:26:09.136641       1 log.go:181] FLAG: --egress-selector-config-file=""
I1215 09:26:09.136646       1 log.go:181] FLAG: --enable-analytics="false"
I1215 09:26:09.136651       1 log.go:181] FLAG: --enable-swagger-ui="false"
I1215 09:26:09.136655       1 log.go:181] FLAG: --help="false"
I1215 09:26:09.136660       1 log.go:181] FLAG: --http2-max-streams-per-connection="1000"
I1215 09:26:09.136665       1 log.go:181] FLAG: --kubeconfig=""
I1215 09:26:09.136670       1 log.go:181] FLAG: --kubeconfig-file=""
I1215 09:26:09.136675       1 log.go:181] FLAG: --log-flush-frequency="5s"
I1215 09:26:09.136684       1 log.go:181] FLAG: --log_backtrace_at=":0"
I1215 09:26:09.136688       1 log.go:181] FLAG: --log_dir=""
I1215 09:26:09.136693       1 log.go:181] FLAG: --logtostderr="true"
I1215 09:26:09.136699       1 log.go:181] FLAG: --profiling="true"
I1215 09:26:09.136705       1 log.go:181] FLAG: --qps="1e+06"
I1215 09:26:09.136713       1 log.go:181] FLAG: --requestheader-allowed-names="[]"
I1215 09:26:09.136718       1 log.go:181] FLAG: --requestheader-client-ca-file=""
I1215 09:26:09.136729       1 log.go:181] FLAG: --requestheader-extra-headers-prefix="[x-remote-extra-]"
I1215 09:26:09.136735       1 log.go:181] FLAG: --requestheader-group-headers="[x-remote-group]"
I1215 09:26:09.136741       1 log.go:181] FLAG: --requestheader-username-headers="[x-remote-user]"
I1215 09:26:09.136746       1 log.go:181] FLAG: --resync-period="10m0s"
I1215 09:26:09.136750       1 log.go:181] FLAG: --secure-port="8443"
I1215 09:26:09.136755       1 log.go:181] FLAG: --stderrthreshold="0"
I1215 09:26:09.136760       1 log.go:181] FLAG: --tls-cert-file="/var/serving-cert/tls.crt"
I1215 09:26:09.136768       1 log.go:181] FLAG: --tls-cipher-suites="[]"
I1215 09:26:09.136774       1 log.go:181] FLAG: --tls-min-version=""
I1215 09:26:09.136778       1 log.go:181] FLAG: --tls-private-key-file="/var/serving-cert/tls.key"
I1215 09:26:09.136783       1 log.go:181] FLAG: --tls-sni-cert-key="[]"
I1215 09:26:09.136789       1 log.go:181] FLAG: --use-kubeapiserver-fqdn-for-aks="false"
I1215 09:26:09.136793       1 log.go:181] FLAG: --v="10"
I1215 09:26:09.136798       1 log.go:181] FLAG: --vmodule=""
I1215 09:26:09.136818       1 run.go:43] Starting kubed...
I1215 09:26:09.543819       1 lib.go:145] Kubernetes version: &version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:09:57Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
I1215 09:26:09.550220       1 operator.go:52] configuring kubed ...
I1215 09:26:09.550274       1 operator.go:52] configuring kubed ...
I1215 09:26:09.552423       1 secure_serving.go:123] Serving securely on [::]:8443
E1215 09:26:18.879506       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:18.879904       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:18.879954       1 errors.go:77] the server could not find the requested resource
E1215 09:26:18.879997       1 errors.go:77] the server could not find the requested resource
E1215 09:26:28.884878       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:28.885511       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:28.885524       1 errors.go:77] the server could not find the requested resource
E1215 09:26:28.885591       1 errors.go:77] the server could not find the requested resource
E1215 09:26:38.875865       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:38.875959       1 errors.go:77] the server could not find the requested resource
E1215 09:26:38.876466       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:38.876583       1 errors.go:77] the server could not find the requested resource
E1215 09:26:38.883618       1 webhook.go:196] Failed to make webhook authorizer request: the server could not find the requested resource
E1215 09:26:38.883739       1 errors.go:77] the server could not find the requested resource
I1215 09:26:38.896352       1 operator.go:96] Stopping kubed controller
```

It works fine after disabling health check.

**Comment by jingzhao-self** [comment]
I also face the same problem in use kubed v1.12.0 in kubernetes v1.22.6
It seems kubed process could not start and could not return http code 200 with "ok".

Don't know how to fix it because it didn't have enough logs that let's us check things.
But I upgrade the version of kubed to v1.13.2 and it's ok now.

**Comment by KannanThiru** [comment]
Issue fixed after upgrading kubed to 0.13.2.
